src/deterministic_vi/NewtonMaximize.jl

```diff
@@ -169,4 +176,42 @@ function from_vector!(sources, x)
     return sources
 end
 
+#######################
+# maximize_two_steps! #
+#######################
+
+function star_only_config(ea::ElboArgs; loc_width = 1.0e-4, loc_scale = 1.0, kwargs...)
+    bound = ea.vp[ea.active_sources]
+    n_sources = length(bound)
+    boxes = Vector{Vector{ParameterConstraint{BoxConstraint}}}(n_sources)
+    simplexes = Vector{Vector{ParameterConstraint{SimplexConstraint}}}(n_sources)
+    for src in 1:n_sources
+        u1, u2 = u_ParameterConstraints(bound[src], loc_width, loc_scale)
+        boxes[src] = [
+            u1,
+            u2,
+            ParameterConstraint(BoxConstraint(-1.0, 10.0, 1.0), ids.r1[1]),
+            ParameterConstraint(BoxConstraint(1e-4, 0.10, 1.0), ids.r2[1]),
+            ParameterConstraint(BoxConstraint(-10.0, 10.0, 1.0), ids.c1[:, 1]),
+            ParameterConstraint(BoxConstraint(1e-4, 1.0, 1.0), ids.c2[:, 1])
+        ]
+        simplexes[src] = [
+            ParameterConstraint(SimplexConstraint(0.01/D, 1.0, D), ids.k[:, 1])
+        ]
+    end
+    return Config(ea; constraints = ConstraintBatch(boxes, simplexes), kwargs...)
+end
+
+function maximize_two_steps!{F,T}(f::F, ea::ElboArgs{T},
+                                  cfg_star::Config = star_only_config(ea),
+                                  cfg_both::Config = Config(ea))
+    ea.vp[1][ids.a] = [1, 0]
```

Comment:
@jeff-regier Sounds good, though I think I may have figured out the problem.

Initializing `ea.vp[1][ids.a] = [1, 0]` here, combined with not placing a `SimplexConstraint` on `a` (in `star_only_config` above), is the cause of the `NaN`s. This problem occurs for both of our objective function implementations (`DeterministicVI.elbo` and the FFT version).

The reason I excluded the `a` constraint from `star_only_config` is because the original code did so (i.e., the `a` indices were in the `omitted_ids`). Adding  the `a` constraint back (or not initializing it to `[1, 0]` in the first place) allows the optimization to run without encountering `NaN`s and allows the optimization tests to pass. However, this means that the new code doesn't exactly match the behavior of the old code. I'm not sure if that's significant or not.

I've just pushed a commit which adds the `a` SimplexConstraint; hopefully tests should pass now. If the behavioral change caused by adding the new constraint is significant, let me know and I'll revert that change and just disable the test.